crates/interpreter/src/valid.rs

```diff
@@ -861,6 +862,8 @@ impl<'a, 'm> Expr<'a, 'm> {
     fn end_label(&mut self) -> CheckResult {
         let results_len = self.label().type_.results.len();
         let mut target = self.branch_target(results_len);
+        // Instead of calling stitch(), make sure the side table entry is consistent with the one in
+        // flash.
```

Comment:
Almost. We need to check that `self.label().branch.is_none_or(|x| x == target)`.